tests/simulation/dummy_bolt.cpp

```diff
@@ -0,0 +1,112 @@
+// Copyright 2022 Memgraph Ltd.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt; by using this file, you agree to be bound by the terms of the Business Source
+// License, and you may not use this file except in compliance with the Business Source License.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+// TODO(gitbuda): Document thoroughly all BoltSession methods.
+
+#include <map>
+#include <string>
+#include <vector>
+
+// TODO(gitbuda): This should move to the utils/logging + refactor where needed.
+#include <spdlog/sinks/daily_file_sink.h>
+
+#include "communication/bolt/v1/session.hpp"
+#include "communication/bolt/v1/value.hpp"
+#include "communication/context.hpp"
+#include "communication/v2/server.hpp"
+#include "communication/v2/session.hpp"
+#include "utils/logging.hpp"
+
+struct InterpretRequest {
+  std::string query;
+  std::map<std::string, memgraph::communication::bolt::Value> params;
+};
+
+struct InterpretResponse {
+  std::vector<std::string> header;
+  int query_id;  // qid
+};
+
+struct PullRequest {
+  int batch_size;
+  int query_id;
+};
+
+struct PullResponse {
+  // NOTE: Interpreter is streaming TypedValues into the TypedValueResultStream.
+  // std::vector<TypedValueEquivalent> batch_result;
+  std::map<std::string, memgraph::communication::bolt::Value> summary;
+};
+
+struct SessionData {};
```

Comment:
`SessionData` will contain info about the "`CypherService`" AKA `QueryEngine` instances.